File: WillMoveToOwnRepo/ProjectorRays/src/ProjectorRays.DotNet/director/Scores/Data/RayScoreHeader.cs
namespace ProjectorRays.director.Scores.Data;

internal record RayScoreHeader
{
    public int ActualSize {get; set; }
    public byte UnkA1 {get; set; }
    public byte UnkA2 {get; set; }
    public byte UnkA3 {get; set; }
    public byte UnkA4 {get; set; }
    public int HighestFrame {get; set; }
    public byte UnkB1 {get; set; }
    public byte UnkB2 {get; set; }
    public short SpriteSize {get; set; }
    public byte UnkC1 {get; set; }
    public byte UnkC2 {get; set; }
    public short ChannelCount {get; set; }
    //public short FirstBlockSize { get; set; }
    public int EntryCount { get; internal set; }
    public int EntrySizeSum { get; internal set; }
    public int NotationBase { get; internal set; }
    public int OffsetsOffset { get; internal set; }
    public int HeaderType { get; internal set; }
    public int TotalLength { get; internal set; }
}

File: WillMoveToOwnRepo/ProjectorRays/src/ProjectorRays_Python/director/scores/data/ray_score_header.py
class RayScoreHeader:
    """Holds header values for a score chunk."""
    def __init__(self):
        self.actual_size = 0
        self.unk_a1 = 0
        self.unk_a2 = 0
        self.unk_a3 = 0
        self.unk_a4 = 0
        self.highest_frame = 0
        self.unk_b1 = 0
        self.unk_b2 = 0
        self.sprite_size = 0
        self.unk_c1 = 0
        self.unk_c2 = 0
        self.channel_count = 0
        self.entry_count = 0
        self.entry_size_sum = 0
        self.notation_base = 0
        self.offsets_offset = 0
        self.header_type = 0
        self.total_length = 0
